compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -1017,7 +1018,47 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             return Ok(cycle_result);
         }
 
-        let (result, dep_node) = self.in_task(|this| this.evaluate_stack(&stack));
+        let (result, dep_node) = self.in_task(|this| {
+            let mut result = this.evaluate_stack(&stack)?;
+
+            // fix issue #103563, we don't normalize nested obligations which produced by `TraitDef` candidate (i.e. using bounds on assoc items as assumptions).
+            // because we don't have enough information to normalize these obligations before evaluating. so we will try to normalize the obligation and evaluate again.
+            // we will replace it with new solver in the future.
```

Comment:
why does tidy not complain about the line length here :sweat_smile: please keep it at around 90 :grin: looks fine :+1:

Reply:
thanks, I have updated this.